groupcache.go

```diff
@@ -329,11 +386,11 @@ func (g *Group) Remove(ctx context.Context, key string) error {
 			close(errs)
 		}()
 
-		// TODO(thrawn01): Should we report all errors? Reporting context
-		//  cancelled error for each peer doesn't make much sense.
 		var err error
 		for e := range errs {
-			err = e
+			if e != nil {
+				err = errors.Join(err, e)
+			}
```

Comment:
If there were some errors but the last one yielded nil, the previous code would not return any error